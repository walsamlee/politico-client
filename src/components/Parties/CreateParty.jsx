/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Component } from 'react';

class CreateParty extends Component {
  constructor(props) {
    super(props);

    this.state = {
      image: 'https://res.cloudinary.com/walsam/image/upload/v1559210390/png-icon.png',
      partyname: '',
      parthqaddress: '',
    };
  }

  changeHandler = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  submitHandler = (event) => {
    event.preventDefault();
    const token = localStorage.getItem('token');

    if (token) {
      const partyData = {
        logoUrl: this.state.image,
        name: this.state.partyname,
        hqAddress: this.state.parthqaddress,
      };

      this.props.createParty(partyData, token);
    }
  }

  imageUploadHandler = (event) => {
    const cloudinaryUrl = 'https://api.cloudinary.com/v1_1/walsam/image/upload';
    const cloudinaryUploadPreset = 'w4rdqtby';

    const file = event.target.files[0];

    const imageFormData = new FormData();

    imageFormData.append('file', file);
    imageFormData.append('upload_preset', cloudinaryUploadPreset);

    fetch(cloudinaryUrl, {
      method: 'POST',
      body: imageFormData,
    })
      .then(response => response.json())
      .then((response) => {
        console.log('Uploaded image: ', response.url);
        this.setState({ image: response.url });
      })
      .catch(error => error);
  }

  resetHandler = () => {
    this.setState({
      partyname: '',
      parthqaddress: '',
    });
  };

  render() {
    return (
      <div className="dashboard">
        <div className="signup-form">
          <form id="create-party" name="sign-up" onSubmit={this.submitHandler}>
            <table className="form-table">
              <tr>
                <td className="label" />
                <td>
                  <label id="party-logo-label" htmlFor="party">
                    <input id="party-logo" className="party-logo" type="file" accept="image/*" name="partylogo" onChange={this.imageUploadHandler} />
                  </label>
                </td>
              </tr>
              <tr>
                <td className="label" />
                <td>
                  <input id="party-name" type="text" name="partyname" placeholder="Party Name" required onChange={this.changeHandler} value={this.state.partyname} />
                </td>
              </tr>
              <tr>
                <td className="label" />
                <td>
                  <input id="part-hq-address" type="text" name="parthqaddress" placeholder="HQ Address" required onChange={this.changeHandler} value={this.state.parthqaddress} />
                </td>
              </tr>
              <tr>
                <td />
                <td>
                  <input type="submit" className="btn btn-cart" value="Save" />
                  <input type="reset" className="btn btn-cart" value="Reset" onClick={this.resetHandler} />
                </td>
              </tr>
            </table>
          </form>
        </div>
      </div>
    );
  }
}

export default CreateParty;
