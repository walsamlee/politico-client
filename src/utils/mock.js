const voteData = {
  office: 1,
  candidate: 1,
  voter: 1,
};

const data = {
  candidate: {
    candidates: [
      {
        officeid: 4,
        firstname: 'Admin',
        lastname: 'Admin',
        othername: 'Sam',
        passporturl: 'https://res.cloudinary.com/walsam/image/upload/v1559164438/IMG_5704.jpg',
        userid: 1,
        partyid: 9,
        name: 'Legislative',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
      },
      {
        officeid: 6,
        firstname: 'Hassan',
        lastname: 'Hadji',
        othername: 'Chikaso',
        passporturl: '',
        userid: 2,
        partyid: 8,
        name: 'Chairman',
        logourl: 'https://res.cloudinary.com/walsam/image/upload/v1550612284/zausu6uc1iet2lo4fsqr.png',
      },
    ],
  },
  getOffices: {
    offices: [],
  },
  parties: {
    parties: [],
  },
};

const details = {
  office: 1,
  candidate: 2,
};

const token = '';

const login = {
  login: {
    token: '',
    user: [
      {
        user: {
          isAdmin: '',
          id: 1,
          passportUrl: '',
          email: '',
          name: '',
          phoneNumber: 14141414441,
        },
      },
    ],
  },
};

const partiesData = {
  parties: {
    fetching: '',
    fetched: '',
    parties: [
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'wrrryery',
        id: 1,
        fetching: true,
      },
      {
        logoUrl: 'https://res.cloudinary.com/walsam/image/upload/v1550613166/difd2f30syyz5fynogsz.png',
        name: 'uiyt',
        id: 2,
        fetching: true,
      },
    ],
  },
};

const partyData = {
  id: '',
  image: '',
  partyname: '',
  parthqaddress: '',
};

const profileData = {
  picture: '',
  name: '',
  email: '',
  phone: '',
};

const registerData = {
  preview: '',
  firstname: '',
  lastname: '',
  othername: '',
  telephone: '',
  email: '',
  password: '',
};

const candidate = {
  passporturl: '',
  firstname: '',
  lastname: '',
  name: '',
  logourl: '',
};

const party = {
  id: '',
  name: '',
};

const aspirantData = {
  id: '',
  office: '',
  party: '',
};

const setPartyState = [
  {
    id: 1,
    logoUrl: '',
    name: '',
    hqAddress: '',
  },
];

const mock = {
  voteData,
  data,
  details,
  token,
  login,
  partiesData,
  partyData,
  profileData,
  registerData,
  candidate,
  party,
  aspirantData,
  setPartyState,
};

export default mock;
