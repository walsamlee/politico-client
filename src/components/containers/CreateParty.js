import { connect } from 'react-redux';
import CreateParty from '../Parties/CreateParty';
import partyAction from '../../actions/partyAction';

const mapStateToProps = state => ({
  data: state,
});

const mapDispatchToProps = dispatch => ({
  createParty: (partyData, token) => {
    dispatch(partyAction.createParty(partyData, token));
  },
});

const createNewParty = connect(mapStateToProps, mapDispatchToProps)(CreateParty);

export default createNewParty;
