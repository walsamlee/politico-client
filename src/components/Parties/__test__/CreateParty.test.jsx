import React from 'react';
import { shallow } from 'enzyme';
import mock from '../../../utils/mock';
import CreateParty from '../CreateParty';

const { token, partyData } = mock;

describe('<CreateParty />', () => {
  it('should render CreateParty component', () => {
    const wrapper = shallow(
      <CreateParty
        partyData={partyData}
        token={token}
        createParty={jest.fn()}
        imageFormData={jest.fn()}
      />,
    );
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: '',
        value: '',
        files: [],
      },
    };
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
    wrapper.instance().submitHandler(event);
    wrapper.instance().changeHandler(event);
    wrapper.instance().resetHandler();
  });
});
