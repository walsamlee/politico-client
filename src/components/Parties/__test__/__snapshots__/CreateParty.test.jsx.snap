// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateParty /> should render CreateParty component 1`] = `ShallowWrapper {}`;
